src/helper/HelperStartWayland.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Session process wrapper
+ * Copyright (C) 2021 Aleix Pol Gonzalez <aleixpol@kde.org>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+/**
+ * This application sole purpose is to launch a wayland compositor (first
+ * argument) and as soon as it's set up to launch a client (second argument)
+ */
+
+#include <QCoreApplication>
+#include <QTextStream>
+#include <QProcess>
+#include "WaylandSocketWatcher.h"
+
+bool startProcess(const QString &cmd)
+{
+    QStringList args = QProcess::splitCommand(cmd);
+    const QString program = args.takeFirst();
+
+    auto app = QCoreApplication::instance();
+    QProcess* p = new QProcess(app);
+    QObject::connect(app, &QCoreApplication::aboutToQuit, p, [p] {
+        p->terminate();
+        if (!p->waitForFinished()) {
+            p->kill();
+        }
+    });
+    p->start(program, args);
+    if (!p->waitForStarted()) {
+        QTextStream(stderr) << "Failed to start: " << cmd << ". " << p->errorString() << "\n";
+        return true;
+    }
+    return true;
+}
+
+int main(int argc, char** argv)
+{
+    QCoreApplication app(argc, argv);
+    if (argc != 3) {
+        QTextStream(stderr) << "Wrong number of arguments\n";
+        return 33;
+    }
+    const QString compositor = app.arguments()[1];
+    const QString client = app.arguments()[2];
+
+    auto watcher = new SDDM::WaylandSocketWatcher(&app, QProcessEnvironment::systemEnvironment());
```

Comment:
Set `WAYLAND_SOCKET` to `sddm-wayland-$SEAT` here or where the helper is called.
Here is probably better as you can set it only when the compositor process is spawned.